newrelic-agent/src/test/java/com/newrelic/agent/SegmentTest.java

```diff
@@ -1140,6 +1210,65 @@ public void setHeader(String name, String value) {
         assertEquals(segmentSpanEvent.getGuid(), secondRootTracerSpanEvent.getParentId());
     }
 
+    @Test
+    public void testSegmentAddCustomAttributeSync() {
+        Transaction.clearTransaction();
+        Tracer rootTracer = makeTransaction();
```

Comment:
lets finish this tracer for better cleanup

Reply:
It looks like it's taken care of in the @After method

Reply:
I don't see the tracer or segment explicitly cleaned up in the @After method. The transaction is cleared but I think it's still good practice to finish the objects in the test.